xbmc/cores/AudioEngine/Sinks/windows/AESinkFactoryWin32.cpp

```diff
@@ -85,33 +85,47 @@ std::vector<RendererDetail> CAESinkFactoryWin::GetRendererDetails()
     PropVariantInit(&varName);
 
     hr = pEnumDevices->Item(i, pDevice.GetAddressOf());
-    EXIT_ON_FAILURE(hr, "Retrieval of WASAPI endpoint failed.")
+    EXIT_ON_FAILURE(hr, "Retrieval of endpoint failed.")
 
     hr = pDevice->OpenPropertyStore(STGM_READ, pProperty.ReleaseAndGetAddressOf());
-    EXIT_ON_FAILURE(hr, "Retrieval of WASAPI endpoint properties failed.")
+    EXIT_ON_FAILURE(hr, "Retrieval of endpoint properties failed.")
 
     hr = pProperty->GetValue(PKEY_Device_FriendlyName, &varName);
-    EXIT_ON_FAILURE(hr, "Retrieval of WASAPI endpoint device name failed.")
+    EXIT_ON_FAILURE(hr, "Retrieval of endpoint device name failed.")
 
     details.strDescription = KODI::PLATFORM::WINDOWS::FromW(varName.pwszVal);
     PropVariantClear(&varName);
 
     hr = pProperty->GetValue(PKEY_AudioEndpoint_GUID, &varName);
-    EXIT_ON_FAILURE(hr, "Retrieval of WASAPI endpoint GUID failed.")
+    EXIT_ON_FAILURE(hr, "Retrieval of endpoint GUID failed.")
 
     details.strDeviceId = KODI::PLATFORM::WINDOWS::FromW(varName.pwszVal);
     PropVariantClear(&varName);
 
     hr = pProperty->GetValue(PKEY_AudioEndpoint_FormFactor, &varName);
-    EXIT_ON_FAILURE(hr, "Retrieval of WASAPI endpoint form factor failed.")
+    EXIT_ON_FAILURE(hr, "Retrieval of endpoint form factor failed.")
 
     details.strWinDevType = winEndpoints[(EndpointFormFactor)varName.uiVal].winEndpointType;
     details.eDeviceType = winEndpoints[(EndpointFormFactor)varName.uiVal].aeDeviceType;
+    PropVariantClear(&varName);
 
+    /* In shared mode Windows tells us what format the audio must be in. */
+    hr = pProperty->GetValue(PKEY_AudioEngine_DeviceFormat, &varName);
+    if (SUCCEEDED(hr) && varName.blob.cbSize >= sizeof(WAVEFORMATEX))
+    {
+      // This may be a WAVEFORMATEXTENSIBLE but the extra data is not needed.
+      WAVEFORMATEX* pwfx = reinterpret_cast<WAVEFORMATEX*>(varName.blob.pBlobData);
+      details.nChannels = pwfx->nChannels;
+      details.m_samplesPerSec = pwfx->nSamplesPerSec;
+    }
+    else
+    {
+      CLog::LogF(LOGERROR, "Getting DeviceFormat failed ({})", CWIN32Util::FormatHRESULT(hr));
```

Comment:
Maybe not treat as error and provide fallback:

change LOGERROR to LOGDEBUG and use as fallback

```
details.nChannels = 2;
details.m_samplesPerSec = 48000;
```
Anyway is no EXIT_ON_FAILURE here.

Then in `EnumerateDevicesEx` remove `if (detail.nChannels)`, etc.

Reply:
I see what you're saying but that seems unnecessary. Though I never saw it fail there was no fallback before, That's why I didn't make it exit on failure. Fine for the log level though, error is too high.
As far as I know it affects only the displayed devices enumeration and AE ignores that info and opens with whatever format it wants anyway. You can confirm by setting to 0 on purpose.